src/transform/tests/testdata/join-fusion

```diff
@@ -0,0 +1,72 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+cat
+(defsource x [int64 int64])
+----
+ok
+
+cat
+(defsource y [int64 int64])
+----
+ok
+
+build
+(join
+  [(get x)
+   (filter
```

Comment:
can you add a third input to the outer join that involves a 
`(filter <something that is not a join>)` so we know that join fusion properly handles filters around non-joins?

Reply:
thanks!